src/sonic-yang-models/yang-models/sonic-auto_techsupport.yang

```diff
@@ -0,0 +1,113 @@
+module sonic-auto_techsupport {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-auto_techsupport";
+    prefix auto_techsupport;
+
+    import sonic-types {
+        prefix stypes;
+    }
+
+    description "Event Driven Techsupport & CoreDump Mgmt Capability in SONiC OS";
+
+    revision 2021-08-09 {
+        description "First Revision";
+    }
+
+    typedef decimal-repr {
+        type decimal64 {
+            fraction-digits 2;
+            range 0.0..99.99; 
+        }
+    }
+
+    container sonic-auto_techsupport {
+
+        container AUTO_TECHSUPPORT {
+
+                description "AUTO_TECHSUPPORT part of config_db.json";
+                
+                container GLOBAL {
+               
+                    leaf state {
+                        description "Knob to make techsupport invocation event-driven based on core-dump generation";
+                        type stypes:admin_mode;
+                    }
+
+                    leaf rate_limit_interval  {
+                        description "Minimum time in seconds between two successive techsupport invocations. Configure 0 to explicitly disable";
+                        type uint16;
+                    }
+
+                    leaf max_techsupport_limit {
+                        /*
+                        A value between (0,100) should be specified. 
+                        Upto two decimal places will be used in the calculation
+                        The actual value in bytes is calculate based on the available space in the filesystem hosting /var/dump
+                        When the limit is crossed, the older core files are incrementally deleted
+                        */
+                        description "Max Limit in percentage for the cummulative size of ts dumps. No cleanup is performed if the value isn't configured or is 0.0";
+                        type decimal-repr;
```

Comment:
value  is between (0, 100) here. But decimal-repr range is from 0.0 to 99.99, can you clarify?

Reply:
It was intended to be b/w 0.0 to 99.99. As for the comment `(A value between (0,100) should be specified, Upto two decimal places will be used in the calculation) `, I've used open brackets to signify the value is between 0 & 100. and the next comment suggests that upto two decimal places are used. So, i think this is clear. Although, let me know, if i have to re-write comments to make them clear

Reply:
Open brackets does not include 0 and 100? Please update comment.

Reply:
Updated the comment